src/components/overview/ProductInformation/AverageRating.jsx

```diff
@@ -0,0 +1,26 @@
+import React, { useState, useEffect } from 'react';
+import StarRating from '../../shared/StarRating.jsx';
+
+const AverageRating = ({state, updateState}) => {
+  let ratingsSum = 0;
+  let reviewCount = 0;
+  for (let key in state.rating) {
+    ratingsSum += parseInt(state.rating[key]) * key;
+    reviewCount += parseInt(state.rating[key]);
+  }
+  const avgRating = Math.round(ratingsSum / reviewCount * 10) / 10;
```

Comment:
For the average rating, what is the reason for rounding down the sum divided by review count? Wouldn't we want the resulting number without rounding? The * 10 and then / 10 also seems redundant. 

I bring this up because your star rating formula and mine need to match, and with the rounding, I think they will be off. 

Take for example the following ratings object: 
```
 "2": "1",
 "3": "2",
"4": "2"
```
the ratingsSum would be 16 and the review count would be 5. 
Currently, your formula resolves to:  round(0.32) / 10 = 0/10 = 0
And mine resolves to: 16 / 5 = 3.2 

Is that right? It seems the latter would be the fraction we want.  
Please let me know if I am mistaken!

Reply:
I think you have order of operations mixed up. Javascript goes from left to right and doesn't prioritize multiplication over division, so it actually resolves to 
```
round(16 / 5 * 10) / 10;
round(3.2 * 10) / 10;
round(32) / 10;
3.2;
```
The * 10 followed by / 10 is to ensure that the resulting number always rounds to no more than one decimal point.

Reply:
Ahh that makes a lot more sense. I was definitely multiplying 5 * 10 first... too early for math. That makes sense. In that case our fractions should be pretty similar but I can refactor mine to keep the decimals small. You're right, a couple of mine have 10+ digits.. ha

One more thing- In the event of a ratings object with 0 ratings (which do exist), we get `0 / 0 / = NaN` but this should be fine because the StarRating component handles falsy values, correct?

Reply:
I confirmed that this works the same as the review's rating star and related product's rating star. I think we can definitely optimize this during next week after implementing our feature all first!

And I just checked it's working fine, even when the NaN is given to the StarRating Component.

Reply:
Awesome. Thanks for checking!